src/source/bcm2835_stream.cpp

```diff
@@ -28,17 +30,86 @@ void Stream::begin(unsigned long baud, int flags)
 	}
 	fcntl(fd, F_SETFL, O_RDWR);
 
-	// Use 8 data bit, no parity and 1 stop bit
+
+	speed_t myBaud;
+	switch ( baud )
+	{
+		case      50:	myBaud =      B50; break;
+		case      75:	myBaud =      B75; break;
+		case     110:	myBaud =     B110; break;
+		case     134:	myBaud =     B134; break;
+		case     150:	myBaud =     B150; break;
+		case     200:	myBaud =     B200; break;
+		case     300:	myBaud =     B300; break;
+		case     600:	myBaud =     B600; break;
+		case    1200:	myBaud =    B1200; break;
+		case    1800:	myBaud =    B1800; break;
+		case    2400:	myBaud =    B2400; break;
+		case    4800:	myBaud =    B4800; break;
+		case    9600:	myBaud =    B9600; break;
+		case   19200:	myBaud =   B19200; break;
+		case   38400:	myBaud =   B38400; break;
+		case   57600:	myBaud =   B57600; break;
+		case  115200:	myBaud =  B115200; break;
+		case  230400:	myBaud =  B230400; break;
+		case  460800:	myBaud =  B460800; break;
+		case  500000:	myBaud =  B500000; break;
+		case  576000:	myBaud =  B576000; break;
+		case  921600:	myBaud =  B921600; break;
+		case 1000000:	myBaud = B1000000; break;
+		case 1152000:	myBaud = B1152000; break;
+		case 1500000:	myBaud = B1500000; break;
+		case 2000000:	myBaud = B2000000; break;
+		case 2500000:	myBaud = B2500000; break;
+		case 3000000:	myBaud = B3000000; break;
+		case 3500000:	myBaud = B3500000; break;
+		case 4000000:	myBaud = B4000000; break;
+
+		default:
+			printf("[ERROR] UART invalid baud: %ld for port: %s\n", baud, port);
+			return;
+	}
+
+
+	struct termios options;
+	memset( &options, 0, sizeof(options) );
+
 	tcgetattr(fd, &options);
-	options.c_cflag = CS8 | CLOCAL | CREAD | CBAUDEX;
-	options.c_cflag &= ~CBAUD;	// use the extended baud
-	options.c_cflag &= ~PARENB;	// no parity
-	options.c_cflag &= ~CSTOPB;	// 1 stop bit
-	options.c_iflag = IGNPAR;
-	options.c_oflag = baud;
-	options.c_lflag = baud;
-	tcflush(fd, TCIFLUSH);
+
+	cfmakeraw( &options );

```

Comment:
Many of the following flags are already set by this function or do not need to be explicitly cleared.
They could be left for verbosity but I haven't decided which way would be better.

Reply:
I think it is best to leave it explicit. If any stability issues arise, it would be easier to tinker. So I am keeping this if it is OK?

Reply:
It's okey as it increases readability and shows the intent.